src/core/entities/crux-spec.ts

```diff
@@ -41,16 +42,30 @@ export class Validations {
     }
     public static validateNameServiceConfig = (nameServiceConfig: IBlockstackServiceInputOptions) => {
         // TODO: domain name validation
-        Validations.validateURL(nameServiceConfig.gaiaHub);
-        Validations.validateURL(nameServiceConfig.subdomainRegistrar);
-        nameServiceConfig.bnsNodes.forEach(Validations.validateURL);
+        if (nameServiceConfig.gaiaHub) {
+            Validations.validateURL(nameServiceConfig.gaiaHub);
+        }
+        if (nameServiceConfig.subdomainRegistrar) {
+            Validations.validateURL(nameServiceConfig.subdomainRegistrar);
+        }
+        if (nameServiceConfig.bnsNodes) {
+            nameServiceConfig.bnsNodes.forEach(Validations.validateURL);
+        }
+    }
+    public static validateAddressMap = (addressMap: IAddressMapping) => {
+        for (const assetId in addressMap) {
+            if (addressMap.hasOwnProperty(assetId)) {
```

Comment:
looks like redundant line

Reply:
It is to check whether the value corresponding to that key is not undefined

Reply:
I guess you need to move the addressMap validation code from blockstack-service to this place